From 1ea85a32c0ae20a104badcec807b72ef458d356d Mon Sep 17 00:00:00 2001
From: Andreas Fuchs <andreas.fuchs@sit.fraunhofer.de>
Date: Tue, 25 Sep 2018 09:35:33 +0200
Subject: [PATCH] ESYS: Fix crypto_ossl return code

Fix returncode and thus test of interal hmac_finish function.

Fixes: #1157

Signed-off-by: Andreas Fuchs <andreas.fuchs@sit.fraunhofer.de>
---
 src/tss2-esys/esys_crypto_ossl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/tss2-esys/esys_crypto_ossl.c b/src/tss2-esys/esys_crypto_ossl.c
index 43088f4a..91871de1 100644
--- a/src/tss2-esys/esys_crypto_ossl.c
+++ b/src/tss2-esys/esys_crypto_ossl.c
@@ -448,7 +448,7 @@ iesys_cryptossl_hmac_finish(IESYS_CRYPTO_CONTEXT_BLOB ** context,
     IESYS_CRYPTOSSL_CONTEXT *mycontext =
         (IESYS_CRYPTOSSL_CONTEXT *) * context;
     if (mycontext->type != IESYS_CRYPTOSSL_TYPE_HMAC) {
-        return_error(TSS2_ESYS_RC_BAD_VALUE, "bad context");
+        return_error(TSS2_ESYS_RC_BAD_REFERENCE, "bad context");
     }
 
     if (*size < mycontext->hmac.hmac_len) {
-- 
2.19.2

